### DISCUSSION TITLE
set frontend user to version

### DISCUSSION THREAD

**Discussion by langert** [pimcore/pimcore#13800] [Q&A / Support]
Hi,

haveing a custom firewall for the frontend, how can I achieve to get the frontendUser als lastModified User into the pimcore versioning for objects? Is this possible? Pimcore does only the job for ids of backend users, while setting it with $object->setUserModification({id}).
Any ideas or do I have to add a custom field to each object with a relation to my frontend users?

Best, Timo

**Comment by dvesh3** [comment]
If you have frontend user as objects in admin then you can create reference of the frontend user to backend user by using `user` datatype and based on the system user id you can get frontend user by calling `Service::getObjectsReferencingUser`. see docs https://pimcore.com/docs/pimcore/current/Development_Documentation/Development_Tools_and_Details/Extending_a_Backend_User.html#page_Extending-the-Pimcore-User

**Comment by langert** [comment]
@dvesh3 thank you for the hint. But so in fact I will have to create a pimcore backend user for every Frontend-User and set the Backend-User Relation to the Frontend-User?
